File: java/hamming/src/main/java/Hamming.java
final class Hamming {
    private final String l;
    private final String r;

    Hamming(String leftStrand, String rightStrand) {
        if (leftStrand.length() != rightStrand.length()) {
            throw new IllegalArgumentException("leftStrand and rightStrand must be of equal length.");
        }

        l = leftStrand;
        r = rightStrand;
    }

    int getHammingDistance() {
        int result = 0;

        for (int i = 0; i < l.length(); i++) {
            if (l.charAt(i) != r.charAt(i)) result++;
        }

        return result;
    }
}

File: kotlin/hamming/src/main/kotlin/Hamming.kt
import kotlin.math.*

object Hamming {
    fun compute(s1: String, s2: String): Int =
        if (s1.length == s2.length)
            s1.zip(s2)
                .map({it.first.compareTo(it.second).absoluteValue})
                .sum()
        else
            throw IllegalArgumentException(
                "left and right strands must be of equal length."
            )
}
